bbc/simorgh
# WSTEAM1: Adds handling for error caused by editorial adding unsupported fields to UGC form

Resolves JIRA [number]

Overall changes
======
Adds handling for error caused by editorial adding unsupported fields to UGC form.

Code changes
======
- Removes JSON.parse() since response is an object (in this edge case)
- Adds check that response is an object (in case of further edge cases)

Testing
======
Testing this edge case
- Pull down this PR, cd into ws-nextjs-app and use yarn dev to run locally
- Visit this URL: http://localhost.bbc.com:7081/somali/send/u130092370?renderer_env=test
- Comment out [these lines](https://github.com/bbc/simorgh/blob/aeee95105dcaf82f141f9f8e4042766d6f74e228/ws-nextjs-app/pages/%5Bservice%5D/send/%5Bid%5D/FormContext/index.tsx#L154-L158) to turn off validation (so we can replicate this editorial edge case)
- Do not fill in the form (so there will be validation errors) & press submit
- Should see error screen after 3 seconds

Testing error if cannot reach form API (other edge case)
- Pull down this PR, cd into ws-nextjs-app and use yarn dev to run locally
- Visit this URL (note no BBC.com): http://localhost:7081/somali/send/u130092370?renderer_env=test
- (Uncomment out [these lines](https://github.com/bbc/simorgh/blob/aeee95105dcaf82f141f9f8e4042766d6f74e228/ws-nextjs-app/pages/%5Bservice%5D/send/%5Bid%5D/FormContext/index.tsx#L154-L158) if commented from earlier test to enable validation)
- Fill in the form & submit
- Should see error screen after 3 seconds

Testing form submits successfully
- Pull down this PR, cd into ws-nextjs-app and use yarn dev to run locally
- Visit this URL: http://localhost.bbc.com:7081/somali/send/u130092370?renderer_env=test
- (Uncomment out [these lines](https://github.com/bbc/simorgh/blob/aeee95105dcaf82f141f9f8e4042766d6f74e228/ws-nextjs-app/pages/%5Bservice%5D/send/%5Bid%5D/FormContext/index.tsx#L154-L158) if commented from earlier test to enable validation)
- Fill in the form & submit
- Form should submit and you can check network tab to see the payload

Note: we should consider adding e2e tests for this edge case

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)
